Wobble/Scheduling/TaskHandler.cs

```diff
@@ -71,42 +82,56 @@ public CancellationToken Run(T input, int delay = 0)
                 Source.Cancel();
                 Source.Dispose();
                 Source = new CancellationTokenSource();
-
-                IsCompleted = false;
-                IsCancelled = false;
-                IsRunning = true;
+                TaskRun(Source.Token, input, delay);
             }
 
-            var token = Source.Token;
-            Task.Run(async () =>
+            return Source.Token;
+        }
+
+        /// <summary>
+        ///     This method will run the task asynchronously.
+        /// </summary>
+        /// <param name="token"></param>
+        /// <param name="input"></param>
+        /// <param name="delay"></param>
+        private async void TaskRun(CancellationToken token, T input, int delay = 0)
+        {
+            try
             {
+                token.ThrowIfCancellationRequested();
```

Comment:
If my analysis above is correct then at this point we're always still inside the `lock {}` in the `Run()` method, so the token can never actually be already cancelled, making this line redundant.

Reply:
You are correct, this should be called after the wait sync from the semaphore